backend/src/main/java/co/inventorsoft/scripty/service/ProjectService.java

```diff
@@ -0,0 +1,87 @@
+package co.inventorsoft.scripty.service;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+
+import javax.transaction.Transactional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import co.inventorsoft.scripty.exception.ApplicationException;
+import co.inventorsoft.scripty.model.dto.ProjectDto;
+import co.inventorsoft.scripty.model.entity.Project;
+import co.inventorsoft.scripty.model.entity.User;
+import co.inventorsoft.scripty.repository.ProjectRepository;
+import co.inventorsoft.scripty.repository.UserRepository;
+import lombok.AccessLevel;
+import lombok.experimental.FieldDefaults;
+
+/**
+ * @author lzabidovsky 
+ */
+@Service
+@Transactional
+@FieldDefaults(level = AccessLevel.PRIVATE)
+public class ProjectService {
+
+	@Autowired
+	ProjectRepository projectRepository;
+
+	@Autowired
+	UserRepository userRepository;
+
+	final static String DEFAULT_DESCRIPTION = "";
+	final static boolean DEFAULT_VISIBILITY = true;
+
+	@Value("${path.local.repo}")
+	String pathLocalRepo;
+	@Value("${directory.separator}")
+	String directorySeparator;
+
+	public long saveProject(ProjectDto project, String username) {
+		User user = userRepository.findByEmail(username).orElseThrow(() -> new ApplicationException("There is no " + username, HttpStatus.NOT_FOUND));
+		if (projectRepository.findByNameAndUser(project.getName(), user).isPresent()) {
```

Comment:
You can build query with `exists` keyword to get boolean result from repository [Spring Data docs](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#_supported_query_keywords). It will execute simple count query which is cheaper than loading whole project row(especially with metadata blob) and will serve your purpose for the check